### ISSUE TITLE
Sync with kantive github (dev) workflows

### ISSUE THREAD

**Issue by navidshaikh** [knative-sandbox/kn-plugin-event#29]
This repo is using worklfows not in sync with how they are used for knative repos.
Also its using Mage build.

@cardil : can we align the build/github workflows?

xref: 
 - https://github.com/knative-sandbox/kn-plugin-event/issues/6
 - https://github.com/knative-sandbox/kn-plugin-event/pull/25#issuecomment-852745237
 - https://github.com/knative-sandbox/kn-plugin-event/pull/28
 
/cc @rhuss

**Comment by rhuss** [comment]
See my comment in https://github.com/knative-sandbox/kn-plugin-event/pull/27#issuecomment-855631060 , in short, my take is:

* All scripts should work without change in the release process as described in https://github.com/knative/release/blob/main/README.md
* wrt/ dependency management should align with the client so that it is easy to inline this plugin (e.g. using vendoring, aligning shared dependencies etc.). We should write a doc, that collects these concrete requirements for a plugin for being able to be inlined. (@dsimansk or do we already have something like this ?)
* Otherwise, I'm not dogmatic if something new is tried out. However, one should always be respectful to other Knative developers, every new way to build a project has a learning cost that might easily outweigh the (often subjective) benefit of an alternate way of building stuff. Especially for a constrained domain like a plugin, it might be questionable to introduce a completely new build system (the project is not about educating how to properly build). Occasional and pass-by users of this repo will be annoyed if they have to learn a new tool if they "just" want to build.

**Comment by cardil** [comment]
I agree that project structure within knative organization needs to be the same - building, CI, releasing etc. The knative-sandbox is, by name, a sandbox, though. That should mean developers should be able to experiment here. Knative-sandbox should allow some level of freedom, and at the same time enforce some rules on projects. But, those should be inclusive to experimatation.

I really think we should allow more diversity, especially in knative-sandbox. For example, there's no reason why there might not be a kn plugin written in rust or c++, as long as the binary works well. Of course, that will not allow to inline the plugin, but that shouldn't force developers to use Golang and follow knative current way of development. That's true for knative-sandbox organization, in my mind. 

Changing a build tool is a much smaller divergence, than using a different programming language. I think knative-sandbox should allow for experiments on that level - development implementations.

Based on that, I think syncing workflows and dependencies is reasonable, but I feel internal opposition to the need for uniformity in project build tools.
